lib/phoenix/channel.ex

```diff
@@ -338,13 +338,39 @@ defmodule Phoenix.Channel do
     Server.push(transport_pid, topic, event, message, socket.serializer)
   end
 
+  @doc """
+  Replies asynchronously to a socket push.
+
+  Useful when you need to reply to a push that can't otherwise be handled using
+  the `{:reply, {status, payload}, socket}` return from your `handle_in`
+  callbacks. `reply/3` will be used in the rare cases you need to perform work in
+  another process and reply when finished by providing the push's `socket.ref`.
+
+  ## Examples
+
+      def handle_in("work", payload, socket) do
+        Worker.perform(payload, socket.ref)
+        {:noreply, socket}
+      end
+
+      def handle_info({:work_complete, result, ref}, socket) do
+        reply socket, ref, {:ok, result}
+        {:noreply, socket}
+      end
+
+  """
+  def reply(socket, ref, {status, payload}) do
```

Comment:
My reservation is that we should not depend on the socket. The socket has the whole assigns which we don't want to be passing around. Thoughts?